src/plots/plots.js

```diff
@@ -659,6 +659,41 @@ plots.linkSubplots = function(newFullData, newFullLayout, oldFullData, oldFullLa
     }
 };
 
+// This function clears defaults between the first and second pass of
+// supplyDefaults. It exists because otherwise null attributes are
+// supplyDefault'd and inherited as *colors* instead of an actual null
+// attribute which needs to be supplydefaulted by the individual
+// expanded traces.
+plots.clearExpandedTraceDefaultColors = function(expandedTraces) {
+    var colorAttrs, path, trace, i, j;
+
+    // A better check *might* be to explicitly check for a groupby transform
+    if(expandedTraces.length <= 1) return;
+
+    function locateExpandedTraceAttrs(attr, attrName, attrs, level) {
+        path[level] = attrName;
+        path.length = level + 1;
+        if(attr.useExpandedTraceDefaultColor) {
+            colorAttrs.push(path.join('.'));
+        }
+    }
+
+    for(i = 0; i < expandedTraces.length; i++) {
+        trace = expandedTraces[i];
+        colorAttrs = [];
+        path = [];
+
+        if(!trace || !trace._module) continue;
+
+        PlotSchema.crawl(trace._module.attributes, locateExpandedTraceAttrs);
```

Comment:
would it be worthwhile to cache `colorAttrs` in `_module` so we don't need to crawl the schema with every `supplyDefaults`?